sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -161,6 +161,30 @@ export class BatchingReceiver extends MessageReceiver {
         }
       };
 
+      // Use new message wait timer only in peekLock mode
+      if (this.receiveMode === ReceiveMode.peekLock) {
+        /**
+         * Resets the timer when a new message is received. If no messages were received for
+         * `newMessageWaitTimeoutInSeconds`, the messages received till now are returned. The
+         * receiver link stays open for the next receive call, but doesnt receive messages until then
+         */
+        this.resetTimerOnNewMessageReceived = () => {
+          if (this._newMessageReceivedTimer) clearTimeout(this._newMessageReceivedTimer);
+          if (this.newMessageWaitTimeoutInSeconds) {
+            this._newMessageReceivedTimer = setTimeout(async () => {
+              const msg =
+                `BatchingReceiver '${this.name}' did not receive any messages in the last ` +
+                `${this.newMessageWaitTimeoutInSeconds} seconds. ` +
+                `Hence ending this batch receive operation.`;
+              log.error("[%s] %s", this._context.namespace.connectionId, msg);
+              finalAction();
+            }, this.newMessageWaitTimeoutInSeconds * 1000);
+          }
+        };
+      } else {
+        this.resetTimerOnNewMessageReceived = () => {};
```

Comment:
Once a batching receiver is created, the receive mode does not change.
The `resetTimerOnNewMessageReceived` is set to an empty function in the constructor.

So, do we need to reset it to empty function in the else block this way?

Reply:
updated